File: Excersise_2b/exercise_2b.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
	int n, cnt=0;
	cin>>n;
	vector<int> arr(n), res;
	for(int i=0 ; i<n ; i++){
		cin>>arr[i];
	}
	for(int i=0 ; i<n ; i++){
		res.push_back(arr[i]);
		if(arr[i]==1){
			cnt++;
		}
		else{
			cnt=0;
		}
		if(cnt==5){
			res.push_back(0);
			cnt=0;
		}
	}
	for(int i=0 ; i<res.size() ; i++){
		cout<<res[i]<<"";
	}
}

File: Exercise_2b/exercise_2b.py
n = int(input())
s = input()
c = 0
res = ""
for i in s:
    if i == '1' and c < 5:
        res += '1'
        c += 1
    elif i == ' ':
        pass
    else :
        res += i
        c = 0
    if c == 5:
        res += '0'
        c = 0
print(res)
